extensions/worker-specific-hashrate-tracking.md

```diff
@@ -0,0 +1,105 @@
+# Stratum V2 Extension: Worker Specific Hashrate Tracking
+
+## 0. Abstract
+
+This document proposes a Stratum V2 extension to enable mining pools to track individual workers (`user_identity`) within extended channels. By including a unique `user_id` field in share messages, pools can track worker-specific hash rate even when multiple devices share a single extended channel.
```

Comment:
Correct. I didn't mention tProxy or other proxies, because I think this extension should be application-agnostic, but I'm up to reconsider that, adding concrete examples

Reply:
I think we could have a FAQ page on the website that can answer common questions. I doubt a dev implementing this would ask: "How can  I track worker-specific hash rate even when multiple devices share a single extended channel." they'll likely ask "How can I track individual worker hash-rate through tProxy."

Will link to this convo in FAQ collection doc I'll start working on. Don't think there needs to be any action on your end atm.

Reply:
Really like this idea 🦾